### ISSUE TITLE
Copy-AlSourceFiles error with Interface type

### ISSUE THREAD

**Issue by g-franck** [microsoft/navcontainerhelper#837]
Hi,

Create-AlProjectFolderFromBCContainer, who calls Copy-AlSourceFiles:

```
Create-AlProjectFolderFromBCContainer -containerName $containerName `
                                      -alProjectFolder $alProjectFolder `
                                      -useBaseLine `
                                      -alFileStructure $alFileStructure
```

throw error below:

```
AVERTISSEMENT : Unexpected error while handling LineWithPrice.Interface.al
Exception lors de l'appel de « Substring » avec « 2 » argument(s) : « La longueur ne peut pas être inférieure à zéro.
Nom du paramètre : length »
Au caractère C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.4.26\AppHandling\Copy-AlSourceFiles.ps1:73 : 29
+ ...                           $id = $line.SubString(0,$line.IndexOf(' '))
+                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : ArgumentOutOfRangeException
```

On Interface type.

### Context:
navcontainerhelper 0.6.4.26
$imageName = "bcinsider.azurecr.io/bcsandbox-master:base-ltsc2019"

**Comment by freddydk** [comment]
Just tried to repro - cannot see any interfaces in the base app.
Could you share the full script?
How did you create the container?
Is it your own baseapp?

**Comment by g-franck** [comment]
Hi Freddy,

You can use this script to reproduce the error:

```
$user = "admin"
$password = "P@ssword1"
$securePassword = (ConvertTo-SecureString -String $password -AsPlainText -Force)
$credential = New-Object PSCredential $user, $securePassword

$containerName = "test"
$auth = "UserPassword"
$imageName = "bcinsider.azurecr.io/bcsandbox-master:base-ltsc2019"
$licenseFile = "C:\Temp\myLicense.flf"
$shortcuts = "Desktop"

New-NavContainer -accept_eula `
                 -containerName $containerName `
                 -auth $auth `
                 -imageName $imageName `
                 -updateHosts `
                 -licenseFile $licenseFile `
                 -credential $credential `
                 -includeAL `
                 -assignPremiumPlan `
                 -shortcuts $shortcuts

$alFileStructure = { Param ([string] $type, [int] $id, [string] $name) 
    switch ($type) {
        "codeunit"          { "src\Codeunit\$($name).al" }
        "controladdin"      { "src\Controladdin\$($name).al" }
        "dotnet"            { "src\dotnet\$($name).al" }
        "enum"              { "src\Enum\$($name).al" }
        "enumextension"     { "src\Enumext\$($name).al" }
        "interface"         { "src\Interface\$($name).al" }
        "page"              { "src\Page\$($name).al" }
        "pagecustomization" { "src\Pagecust\$($name).al" }
        "pageextension"     { "src\Pageext\$($name).al" }
        "profile"           { "src\profile\$($name).Profile.al" }
        "query"             { "src\Query\$($name).al" }
        "report"            { "src\Report\$($name).al" }
        ".rdlc"             { "src\Report\$name$type" }
        ".docx"             { "src\Report\$name$type" }
        "requestpage"       { "src\Report\$($name).al" }
        "table"             { "src\Table\$($name).al" }
        "tableextension"    { "src\Tableext\$($name).al" }
        "xmlport"           { "src\XmlPort\$($name).al" }
        ".xlf"              { "translations\$name$type" }
        default             { "$($name).$($type).al" }
    }
}

$alProjectFolder = "C:\ProgramData\NavContainerHelper\Extensions\$containerName\AL\BaseApp"
Create-AlProjectFolderFromNavContainer -containerName $containerName `
                                       -alProjectFolder $alProjectFolder `
                                       -alFileStructure $alFileStructure `
                                       -useBaseLine
```

**Output:**

```
NavContainerHelper is version 0.6.4.26
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Entreprise - 1909
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Using image bcinsider.azurecr.io/bcsandbox-master:base-ltsc2019
Creating Container test
Version: 16.0.11007.0-W1
Style: sandbox
Platform: 16.0.10979.0
Generic Tag: 0.0.9.98
Container OS Version: 10.0.17763.973 (ltsc2019)
Host OS Version: 10.0.18363.592 (1909)
The container operating system does not match the host operating system, forcing hyperv isolation.
Using locale en-US
Using hyperv isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\Temp\myLicense.flf
Files in C:\ProgramData\NavContainerHelper\Extensions\test\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container test from image bcinsider.azurecr.io/bcsandbox-master:base-ltsc2019
126d9c544503b0a460825e6b53680a684a5ebc19db8a5ff2afd9bbd45b8220e7
Waiting for container test to be ready
Initializing...
Setting host.docker.internal to 192.168.1.15 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.1.15 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.31.32.1 in container hosts file
Starting Container
Hostname is test
PublicDnsName is test
WARNING: DNS resolution not working from within the container.
WARNING: Container starts with TimeZone = Paris, Madrid, which is not recognized in the list of TimeZones.
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint D9197D919E3BC2FF0031EAB507B2A6DAA5447B5F
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
WARNING: This license is not compatible with this version of Business Central.
Creating SUPER user
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
Assign Premium plan for ADMIN
Container IP Address: 172.31.32.223
Container Hostname  : test
Container Dns Name  : test
Web Client          : http://test/BC/
Dev. Server         : http://test
Dev. ServerInstance : BC
Setting test to 172.31.32.223 in host hosts file

Files:
http://test:8080/ALLanguage.vsix

Initialization took 262 seconds
Ready for connections!
Reading CustomSettings.config from test
Creating Desktop Shortcuts for test
Extracting C:\Applications\BaseApp\Source\Base Application.Source.zip
Creating .net Assembly Reference Folder for VS Code
Copying DLLs from C:\Windows\assembly to assemblyProbingPath
Copying DLLs from C:\Program Files\Microsoft Dynamics NAV\160\Service to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Microsoft Dynamics NAV\160\RoleTailored Client to assemblyProbingPath
Copying DLLs from C:\Test Assemblies\Mock Assemblies to assemblyProbingPath
Copying DLLs from C:\Program Files (x86)\Open XML SDK to assemblyProbingPath
Container test successfully created
Copying Al Source Files from C:\ProgramData\NavContainerHelper\Extensions\Original-16.0.11007.0-W1-al\* to C:\ProgramData\NavContainerHelper\Extensions\test\AL\BaseApp
AVERTISSEMENT : Unexpected error while handling LineWithPrice.Interface.al
Exception lors de l'appel de « Substring » avec « 2 » argument(s) : « La longueur ne peut pas être inférieure à zéro.
Nom du paramètre : length »
Au caractère C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.4.26\AppHandling\Copy-AlSourceFiles.ps1:73 : 29
+ ...                           $id = $line.SubString(0,$line.IndexOf(' '))
+                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : ArgumentOutOfRangeException
```

Cordially

**Comment by freddydk** [comment]
Yep - came to the same conclusion.
thanks

**Comment by freddydk** [comment]
shipped in 0.6.4.27
